test/process_bootstrap_test.cc

```diff
@@ -0,0 +1,1153 @@
+                                                         "--stats-flush-interval 20 "
+                                                         "--tls-context {} "
+                                                         "https://www.example.org",
+                                                         stats_sink_json, tls_context_json));
+
+  absl::StatusOr<Bootstrap> expected_bootstrap = parseBootstrapFromText(R"pb(
+    static_resources {
+      clusters {
+        name: "0"
+        type: STATIC
+        connect_timeout {
+          seconds: 30
+        }
+        max_requests_per_connection {
+          value: 4294937295
+        }
+        circuit_breakers {
+          thresholds {
+            max_connections {
+              value: 100
+            }
+            max_pending_requests {
+              value: 10
+            }
+            max_requests {
+              value: 100
+            }
+            max_retries {
+            }
+          }
+        }
+        transport_socket {
+          name: "envoy.transport_sockets.tls"
+          typed_config {
+            [type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext] {
+              common_tls_context {
+                tls_params {
+                  cipher_suites: "-ALL:ECDHE-RSA-AES256-GCM-SHA384"
+                }
+                alpn_protocols: "http/1.1"
+              }
+              sni: "www.example.org"
+            }
+          }
+        }
+        load_assignment {
+          cluster_name: "0"
+          endpoints {
+            lb_endpoints {
+              endpoint {
+                address {
+                  socket_address {
+                    address: "127.0.0.1"
+                    port_value: 443
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    stats_sinks {
+      name: "envoy.stat_sinks.statsd"
+      typed_config {
+        [type.googleapis.com/envoy.config.metrics.v3.StatsdSink] {
+          tcp_cluster_name: "statsd"
+        }
+      }
+    }
+    stats_flush_interval {
+      seconds: 20
+    }
+  )pb");
+  ASSERT_THAT(expected_bootstrap, StatusIs(absl::StatusCode::kOk));
+
+  absl::StatusOr<Bootstrap> bootstrap =
+      createBootstrapConfiguration(*options, uris_, request_source_uri_, number_of_workers_);
+  ASSERT_THAT(bootstrap, StatusIs(absl::StatusCode::kOk));
+  EXPECT_THAT(*bootstrap, EqualsProto(*expected_bootstrap));
+
+  // Ensure the generated bootstrap is valid.
+  Envoy::MessageUtil::validate(*bootstrap, Envoy::ProtobufMessage::getStrictValidationVisitor());
+}
+
+TEST_F(CreateBootstrapConfigurationTest, CreatesBootstrapSetsMaxRequestToAtLeastOne) {
+  uris_.push_back(std::make_unique<UriImpl>("http://www.example.org"));
+  resolveAllUris();
+
+  std::unique_ptr<Client::OptionsImpl> options = Client::TestUtility::createOptionsImpl(
+      "nighthawk_client --max-pending-requests 0 http://www.example.org");
```

Comment:
nit: add a comment to highlight that even though the CLI flag has value 0, created bootstrap will still be set to 1. (otherwise it is not obvious)

Reply:
Done in 37a2d10d464f9d197183ba869ed62a7de0dc4335.